test/e2e/common.go

```diff
@@ -82,3 +88,48 @@ func GetLatestReleaseOfMinor(ctx context.Context, minorRelease string) (string,
 	releaseMarker := fmt.Sprintf(latestReleaseMarkerPrefix, minorRelease)
 	return clusterctl.ResolveRelease(ctx, releaseMarker)
 }
+
+// verifyV1Beta2Conditions checks the Cluster and Machines of a Cluster that
+// the given v1beta2 condition types are set to true without a message, if they exist.
+func verifyV1Beta2Conditions(ctx context.Context, c client.Client, clusterName, clusterNamespace string, v1beta2conditionTypes map[string]struct{}) {
+	cluster := &clusterv1.Cluster{}
+	key := client.ObjectKey{
+		Namespace: clusterNamespace,
+		Name:      clusterName,
+	}
+	Eventually(func() error {
+		return c.Get(ctx, key, cluster)
+	}, 3*time.Minute, 10*time.Second).Should(Succeed(), "Failed to get Cluster object %s", klog.KRef(clusterNamespace, clusterName))
+
+	if len(cluster.Status.Conditions) > 0 {
+		for _, condition := range cluster.Status.Conditions {
+			if _, ok := v1beta2conditionTypes[condition.Type]; !ok {
+				continue
+			}
+			Expect(condition.Status).To(Equal(metav1.ConditionTrue), "The v1beta2 condition %q on the Cluster should be set to true", condition.Type)
+			Expect(condition.Message).To(BeEmpty(), "The v1beta2 condition %q on the Cluster should have an empty message", condition.Type)
+		}
+	}
```

Comment:
```suggestion
	Eventually(func() error {
		cluster := &clusterv1.Cluster{}
	        key := client.ObjectKey{
		        Namespace: clusterNamespace,
		        Name:      clusterName,
	        }
		if err:= c.Get(ctx, key, cluster); err != nil {
		     return err
		}
                if !conditions.IsTrue(cluster, clusterv1.ClusterInfrastructureReadyCondition) {
		        return fmt.Error("Available condition on the Cluster in not true")
	        }
	        return nil
	}, 1*time.Minute, 10*time.Second).Should(Succeed(), "Available condition on Cluster %s in not true ", klog.KRef(clusterNamespace, clusterName))
```

So we give some time for the cluster condition to get stable (instead of only retrying the get)

Reply:
Same for Machines but checking the ready condition